src/variant_eq.rs

```diff
@@ -0,0 +1,208 @@
+                }
+            }
+            b_dupe.is_empty()
+        }
+        (Variant::Axes(a), Variant::Axes(b)) => a == b,
+        (Variant::BinaryString(a), Variant::BinaryString(b)) => a == b,
+        (Variant::Bool(a), Variant::Bool(b)) => a == b,
+        (Variant::BrickColor(a), Variant::BrickColor(b)) => a == b,
+        (Variant::CFrame(a), Variant::CFrame(b)) => {
+            vector_eq(&a.position, &b.position)
+                && vector_eq(&a.orientation.x, &b.orientation.x)
+                && vector_eq(&a.orientation.y, &b.orientation.y)
+                && vector_eq(&a.orientation.z, &b.orientation.z)
+        }
+        (Variant::Color3(a), Variant::Color3(b)) => {
+            approx_eq!(f32, a.r, b.r) && approx_eq!(f32, a.b, b.b) && approx_eq!(f32, a.g, b.g)
+        }
+        (Variant::Color3uint8(a), Variant::Color3uint8(b)) => a == b,
+        (Variant::ColorSequence(a), Variant::ColorSequence(b)) => {
+            if a.keypoints.len() != b.keypoints.len() {
+                return false;
+            }
+            let mut a_keypoints = Vec::with_capacity(a.keypoints.len());
+            let mut b_keypoints = Vec::with_capacity(b.keypoints.len());
+            for keypoint in &a.keypoints {
+                a_keypoints.push(keypoint)
+            }
+            for keypoint in &b.keypoints {
+                b_keypoints.push(keypoint)
+            }
+            a_keypoints.sort_unstable_by(|k1, k2| k1.time.partial_cmp(&k2.time).unwrap());
+            b_keypoints.sort_unstable_by(|k1, k2| k1.time.partial_cmp(&k2.time).unwrap());
+            for (a_kp, b_kp) in a_keypoints.iter().zip(b_keypoints) {
+                if !(approx_eq!(f32, a_kp.time, b_kp.time)
+                    && approx_eq!(f32, a_kp.color.r, b_kp.color.r)
+                    && approx_eq!(f32, a_kp.color.g, b_kp.color.g)
+                    && approx_eq!(f32, a_kp.color.b, b_kp.color.b))
+                {
+                    return false;
+                }
+            }
+            true
+        }
+        (Variant::Content(a), Variant::Content(b)) => a == b,
+        (Variant::Enum(a), Variant::Enum(b)) => a == b,
+        (Variant::Faces(a), Variant::Faces(b)) => a == b,
+        (Variant::Float32(a), Variant::Float32(b)) => approx_eq!(f32, *a, *b),
+        (Variant::Float64(a), Variant::Float64(b)) => approx_eq!(f64, *a, *b),
+        (Variant::Font(a), Variant::Font(b)) => {
+            a.weight == b.weight
+                && a.style == b.style
+                && a.family == b.family
+                && a.cached_face_id == b.cached_face_id
+        }
+        (Variant::Int32(a), Variant::Int32(b)) => a == b,
+        (Variant::Int64(a), Variant::Int64(b)) => a == b,
+        (Variant::MaterialColors(a), Variant::MaterialColors(b)) => a.encode() == b.encode(),
+        (Variant::NumberRange(a), Variant::NumberRange(b)) => {
+            approx_eq!(f32, a.max, b.max) && approx_eq!(f32, a.min, a.max)
+        }
```

Comment:
```suggestion
        (Variant::NumberRange(a), Variant::NumberRange(b)) => {
            approx_eq!(f32, a.max, b.max) && approx_eq!(f32, a.min, b.min)
        }
```

Reply:
Good catch! Fixed this and cleaned up the variant_eq function a bit more since it'd been 9 months since I changed it.